src/javaharness/java/arcs/webimpl/WebHarnessComponent.java

```diff
@@ -3,7 +3,9 @@
 import arcs.api.ArcsEnvironment;
 import arcs.nativeparticles.NativeParticlesModule;
 import dagger.Component;
+import javax.inject.Singleton;
 
+@Singleton
```

Comment:
Why does this have to be a `Singleton` now, but wasn't before?

Reply:
this change was actually already merged in https://github.com/PolymerLabs/arcs/pull/3295 - without it the `public static Map<String, ArcsEnvironment.DataListener> provideInProgressListeners()` always provides a new map and the listeners are not registered - this was a preexisting problem in the javaharness, and is unrelated to the migration.